### ISSUE TITLE
Enable new Roslyn code analyzer

### ISSUE THREAD

**Issue by jefflill** [nforgeio/neonKUBE#665]
We're seeing lots of `NullReferenceException` and other warnings, presumably from the old FxCop analyzer. I also suspect this is the is the reason why generated code files are getting locked, requiring logout/login.

https://docs.microsoft.com/en-us/visualstudio/code-quality/roslyn-analyzers-overview?view=vs-2019
https://marketplace.visualstudio.com/items?itemName=VisualStudioPlatformTeam.MicrosoftCodeAnalysis2019

**Comment by jefflill** [comment]
I disabled legacy analyzers by adding the following to the ***.csproj** files:
```
<RunCodeAnalysis>false</RunCodeAnalysis>
```
I'm still seeing these warning though:
```
45>CSC : warning AD0001: Analyzer 'Microsoft.AspNetCore.Mvc.Analyzers.TopLevelParameterNameAnalyzer' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.
```

**Comment by jefflill** [comment]
After spending way too much time yesterday and this morning on this, I couldn't get this to work properly.  I disabled the legacy analyzers and installed the nuget packages and got 10.5K+ errors and warnings.  After reading some more, I figured out that it is possible to have a central ruleset file that controls the rule enablement.  I set this up and configured a more reasonable set rules and this kind of worked.

The problem was that individual projects would compile without warning but if I compiled the while solution, warnings would pop back; like two analyzers were fighting or something.

I'm keeping the legacy analyzer disablement (hopefully helping the file locked during build problem) and reverting the other changes.  We'll keep this on the backlog.
